Update Advanced Usage chapter in docs [doc] (#1031)
avaframe/AvaFrame: docs/developinstallwin.rst

  cd [YOURDIR]
  git clone https://github.com/avaframe/AvaFrame.git
  cd AvaFrame

Compile the cython com1DFA part::

   python setup.py build_ext --inplace

.. Warning::
   You will have to do this compilation every time something changes in the cython code. We also suggest
   to do this everytime updates from the repositories are pulled.

   **Before** compilation in Windows, make sure to delete ``AvaFrame/build`` directory, in addition to any .pyd, .c, and
   .pycache files in ``AvaFrame/avaframe/com1DFA``

Install avaframe and its requirements by **either** doing::

  pip install -e .

or if this fails (see `github issue 986 <https://github.com/avaframe/AvaFrame/issues/986>`_), do::

  python setup.py develop

This installs avaframe in editable mode, so every time you import avaframe the
current (local) version will be used.

Test it by starting ``python`` and do an ``import avaframe``. If no error comes
up, you are good to go.

If wanted: Integrate conda environment into editor of your choice (i.e. PyCharm, Spyder or any other)

Head over to :ref:`gettingstarted:First run` for the next steps.

Using QGIS from Conda
^^^^^^^^^^^^^^

> It might be wanted to both have a script installation and the plugin installed on your machine at the same time.
> Unfortunately, installing AvaFrame on Windows via QGIS plugin might overrule your script AvaFrame installation, causing
> issues when trying to work on both versions in parallel. This depends on your mode of installation (direct installer, conda, OSGeo4W...).

Reviewer: More direct and concise:
```suggestion
It is possible to have the script installation and the plugin installed on your machine at the same time.
However, be aware: Depending on your mode of QGis installation (direct installer, conda, OSGeo4W...) your script AvaFrame installation might be overruled.  
```

> Below, we have compiled a **short guide**, describing a way to install QGIS and the Connector plugin in a separate
> *conda* environment. This method will result in two script installations (if the steps above have been followed),
> where one will be used by QGIS and the other may be worked on separately, preventing any overlap.

Reviewer: ```suggestion
The steps below install QGis and the Connector plugin in a separate *conda* environment. If the steps above have been followed, this method will result in two script installations where one will be used by QGis and the other can be worked on separately, preventing any overlap.
```

> Firstly, make sure that if you have another QGIS installation somewhere, that the AvaFrame plugin is not installed.
> Next, create and activate a new environment, which will contain your QGIS installation. The name of the environment
> can be anything, but since we will install the latest version of QGIS, we will call it *qgis_latest* in this example.
> Open your conda terminal and use::

Reviewer: ```suggestion
If you have another QGis installation somewhere, make sure that the AvaFrame plugin is not installed.
To create and activate a new environment, which will contain your installation of QGis open your conda terminal and run::
```

  conda create -n qgis_latest
  conda activate qgis_latest

To **install the latest QGIS version**, use::

  conda install qgis --channel conda-forge

.. Note::

>   Because conda will always try to install the latest version, if you want to use another version, you need to specify it, e.g.:

Reviewer: ```suggestion
   Conda will always try to install the latest version. If you want to use another version, you need to specify it, e.g.:
```

  ``conda install qgis=3.34.11 --channel conda-forge``

Now install avaframe into the qgis environment. This is to make sure that the plugin will find this version first, and install the necessary dependencies.
In the conda terminal, use::

  pip install avaframe

Now find the directory where your environment is located. The path should be something like:
``C:\Users\USER\miniconda3\envs\qgis_latest``. Change into it::

  cd \miniconda3\envs\qgis_latest

.. Note::
  **Tip:** To find the path, you can do ``pip show avaframe`` and copy the

>   first part which shows your environment directory

Reviewer: I think we could leave this out and just say: use a directory that's different then your script installation from above, right? Since then we would not touch any conda env directories. The requirements should come through the commands below anyways...

Author: I am not sure about this one.. I believe it's important to install the requirements before in the environment right? At least we need pip numpy cython. Or will the cloned AF be able to use dependencies outside the environment?

Reviewer: That is a good point, installing avaframe however is not a good idea. Please switch `pip install avaframe` to `conda install pip numpy cython`. The rest of the requirements will be met via the pip install lower down.

Clone the AvaFrame repository and change into it::

  git clone https://github.com/avaframe/AvaFrame.git
  cd AvaFrame

.. Note::
  This will pull the latest developer version of AvaFrame, if you want the current release version for the plugin,
  you need to specify the version, e.g.:

  ``https://github.com/avaframe/AvaFrame/tree/1.8.3``

Compile the cython com1DFA part and create an editable avaframe version in your environment::

  python setup.py build_ext --inplace
  pip install -e .

To **run QGIS**, in the terminal, activate the environment (if not activated already) and run the ``qgis`` command::

  conda activate qgis_latest
  qgis

> 

Reviewer: Mh, I would only write 'Run qgis: ' 

If people followed the guide, the environment should be activated. If not, they installed AvaFrame into a systemwide directory, leading to problems...

Author: Sure, for the installation it's not needed to activate the env again. I was thinking for people just to refer back on how to open qgis again, after pc restart etc., to make sure again to activate the env. Could alternatively move the info downwards..

Reviewer: In my experience it is better to keep the explanations as simple as possible: having the activate a second time here could lead to a 'double stacked' environment, which should not be a problem, but could be.... In the context/flow of this explanation please leave the activate out here. 
And furthermore: I'm hoping that anyone who is attempting this has at least the presence to get familiar with conda in case they don't know it already. So they should be aware of the fact that an environment has to be activated before they can use it again. Especially since QGis won't be found without the env...

Now **install the AvaFrameConnector plugin** via QGIS as per usual (:ref:`installation:Operational Installation`).

AvaFrame should now be installed, and both installations should work and be editable separately. To display the installed version and installation location, use
``pip show avaframe`` in the respective environment, or use the *getVersion* function in the plugin. Note that the script
versions are separate, and changes in one installation will not affect the other.

**Updating QGIS**

To update QGIS to the most recent version, you need to run the following command with the respective environment active::

  conda update qgis -c conda-forge

**Final comments:**

Be aware that the QGIS Conda package does not provide the same user experience as the QGIS installer version.
For example, there are no desktop icons or file association, and it does not include GRASS and SAGA, etc..

On the other hand, the QGIS installation through Conda will share user configurations, installed plugins,

> with any other QGIS installation on your system.

Reviewer: I would leave this out, since I'm not sure that is true for all setups...

Author: This info is from [here](https://plugins.qgis.org/planet/tag/conda/). True, could leave this out though, as I am not completely sure either how QGIS works there..

Reviewer: The info you linked is from 2019 and as you/we are not completely sure how this is handled on current version, please just leave it out...

suggest to work with miniconda/anaconda. For installation see `miniconda
<https://docs.conda.io/en/latest/miniconda.html>`_ or
`anaconda <https://docs.anaconda.com/anaconda/install/linux/>`_.

Install `Microsoft C++ compiler <https://wiki.python.org/moin/WindowsCompilers>`_.
Follow the installation steps for the version corresponding to the installed python version.

Setup AvaFrame
^^^^^^^^^^^^^^

Open conda shell and create a new `conda environment
<https://conda.io/projects/conda/en/latest/user-guide/concepts/environments.html>`_
for AvaFrame, activate it and install pip, numpy and cython in this environment::

  conda create --name avaframe_env
  conda activate avaframe_env
  conda install pip numpy cython

Clone the AvaFrame repository (in a directory of your choice: [YOURDIR]) and change into it::

  cd [YOURDIR]
  git clone https://github.com/avaframe/AvaFrame.git
  cd AvaFrame

Compile the cython com1DFA part::

   python setup.py build_ext --inplace

.. Warning::
   You will have to do this compilation every time something changes in the cython code. We also suggest
   to do this everytime updates from the repositories are pulled.

>    **Note**: After (re)compilation in Windows, delete AvaFrame/build directory, in addition to any .pyd, .c, and .pycache files in AvaFrame/avaframe/com1DFA

Reviewer: ```suggestion
   **Note**: For compilation in Windows, make sure to delete AvaFrame/build directory, in addition to any .pyd, .c, and .pycache files in AvaFrame/avaframe/com1DFA
```

Install avaframe and its requirements by **either** doing::

  pip install -e .

or if this fails (see `github issue 986 <https://github.com/avaframe/AvaFrame/issues/986>`_), do::

  python setup.py develop

This installs avaframe in editable mode, so every time you import avaframe the
current (local) version will be used.

Test it by starting ``python`` and do an ``import avaframe``. If no error comes
up, you are good to go.

If wanted: Integrate conda environment into Editor of your choice (i.e. PyCharm, Spyder or any other)

Head over to :ref:`gettingstarted:First run` for the next steps.

NOTE:

>   Installing AvaFrame on Windows via QGIS plugin writes AvaFrame to system-wide python. This will override any avaframe installation in conda environment!

Reviewer: Mh, this is not entirely true but depends on the mode of installation. There is no such thing as system-wide python on Windows, only python installations that might be accessed/shared. I suggest rewriting it to something like:
```suggestion
  Installing AvaFrame on Windows via QGIS plugin might overrule your script AvaFrame installation! This depends on your mode of installation (conda, OSGeo4W, direct installer...).
```
